src/ndarray/ndarray.cc

```diff
@@ -181,19 +181,20 @@ NDArray NDArray::MKLDNNDataReshape(const TShape &shape) const {
     NDArray ret(shape, ctx(), true, dtype());
     // We shouldn't submit the reorder primitive here because submit will
     // be called in operators.
-    auto format = GetDefaultFormat(ptr_->mkl_mem_->get_primitive_desc().desc());
-    CHECK_NE(format, ptr_->mkl_mem_->get_primitive_desc().desc().data.format);
-    auto def_pd = GetPrimitiveDesc(ptr_->mkl_mem_->get_primitive_desc(), format);
+    auto format = ptr_->mkl_mem_->GetDefaultFormat();
```

Comment:
Can we please reduce the usage of 'auto'?  Usually, 'auto' is best for obvious types or very tedious declarations such as stl map iterators.  For this, there's several calls that I have no idea what the object types are.  It makes the code hard to understand.  There's so many autos that it almost looks like javascript :)